grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationshipAtom.java

```diff
@@ -539,32 +539,27 @@ else if(compatibleTypesFromRoles.isEmpty() || types.isEmpty()){
      * {@link EntityType}s only play the explicitly defined {@link Role}s (not the relevant part of the hierarchy of the specified {@link Role}) and the {@link Role} inherited from parent
      * @return list of {@link RelationshipType}s this atom can have ordered by the number of compatible {@link Role}s
      */
-    public ImmutableList<Type> inferPossibleTypes(Answer sub) {
+    public ImmutableList<Type> inferPossibleTypes() {
         if (getSchemaConcept() != null) return ImmutableList.of(getSchemaConcept().asRelationshipType());
         if (possibleRelations == null) {
-            Multimap<RelationshipType, Role> compatibleConfigurations = inferPossibleRelationConfigurations(sub);
+            Multimap<RelationshipType, Role> compatibleConfigurations = inferPossibleRelationConfigurations();
             Set<Var> untypedRoleplayers = Sets.difference(getRolePlayers(), getParentQuery().getVarTypeMap().keySet());
             Set<RelationshipAtom> untypedNeighbours = getNeighbours(RelationshipAtom.class)
                     .filter(at -> !Sets.intersection(at.getVarNames(), untypedRoleplayers).isEmpty())
                     .collect(toSet());
 
             ImmutableList.Builder<Type> builder = ImmutableList.builder();
-            //prioritise relations with higher chance of yielding answers
             compatibleConfigurations.asMap().entrySet().stream()
-                    //prioritise relations with more allowed roles
-                    .sorted(Comparator.comparing(e -> -e.getValue().size()))
-                    //prioritise relations with number of roles equal to arity
-                    .sorted(Comparator.comparing(e -> e.getKey().relates().count() != getRelationPlayers().size()))
```

Comment:
why is this removed? i know that intelliJ doesn't like it but I don't get why

Reply:
IntelliJ said it was redundant and did not affect processing. Apparently java is actually smart enough to optimise these cases by itself. So these lines were (as far as I understand it) useless.

Reply:
IntelliJ probably thinks the sorts aren't stable, so `stream.sorted(byFoo).sorted(byBar)` is the same as `stream.sorted(byBar)`.

@kasper-piskorski if you want to sort by Foo first, then by Bar second try this:

```
stream.sorted(byFoo.thenComparing(byBar))
```

I don't know if it will make a difference, but it won't make things worse and at least IntelliJ will stop complaining 😛

Reply:
From my understanding the sorting here is stable. I have just tested with markit. Removing the extra sorting makes some of the query execution times go through the roof xD